doc/howto/write_an_alphabet/dna2_alphabet.cpp

```diff
@@ -0,0 +1,98 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+//! [writable_alphabet]
+#include <seqan3/alphabet/concept.hpp>                   // alphabet concept checks
+
+struct dna2
+{
+    uint8_t rank{};
+
+    static constexpr size_t alphabet_size = 2;
+
+    uint8_t to_rank() const noexcept
+    {
+        return rank;
+    }
+
+    dna2 & assign_rank(uint8_t const rk) noexcept
+    {
+        assert(rk < alphabet_size);
+        rank = rk;
+        return *this;
+    }
+
+    char to_char() const noexcept
+    {
+        // map 0 => 'S' and 1 => 'W'
+        char const rank_to_char[2] {'S', 'W'};
+        return rank_to_char[rank];
+    }
+
+    dna2 & assign_char(char const ch) noexcept
+    {
+        switch (ch)
+        {
+            case 'W': rank = 1;
+            case 'w': rank = 1; break;                   // allow assignment from uppercase and lowercase
```

Comment:
The first line is superfluous, plus this might trigger a warning. 

Instead put it in one line:

```suggestion
            case 'W': case 'w': rank = 1; break;             // allow assignment from uppercase and lowercase
```